specs/greeting_spec.js

```diff
@@ -1,40 +1,52 @@
 const assert = require('assert')
-const Botmock = require('botkit-mock')
 
+const { getBasicController } = require('./helpers')
 const greetingController = require('../features/greeting')
 const replies = require('../features/greeting/replies.js')
 
 describe('Sample hears controller', () => {
   beforeEach(() => {
-    this.controller = Botmock({})
-    this.bot = this.controller.spawn({ type: 'slack' })
+    this.controller = getBasicController()
     greetingController(this.controller)
   })
 
-  it(
-    'Should return any greeting if user types `hi`',
-    () => this.bot.usersInput([{
+  it('Should return any greeting if user types `hi`', async () => {
+    await this.controller.usersInput([{
+      type: 'message',
+      channel: 'channel',
       messages: [{
         text: 'hi', isAssertion: true,
       }],
     }]).then(message => assert(replies.includes(message.text)))
-  )
+  })
+
+  it('Shouldn\'t return any greeting if user types `Morning` in one line code block', async () => {
+    await this.controller.usersInput([{
+      type: 'message',
+      channel: 'channel',
+      messages: [{
+        text: '`Morning`', isAssertion: true,
+      }],
+    }]).then(message => assert.deepEqual(message, {}))
+  })
 
-  it(
-    "Shouldn't return any greeting if user types hi in one line code block",
-    () =>this.bot.usersInput([{
+  it('Shouldn\'t return any greeting if user types `Morning` in multiline code block', async () => {
+    await this.controller.usersInput([{
+      type: 'message',
+      channel: 'channel',
       messages: [{
-        text: '`hi`', isAssertion: true,
+        text: '```Morning```', isAssertion: true,
       }],
     }]).then(message => assert.deepEqual(message, {}))
-  )
+  })
 
-  it(
-    "Shouldn't return any greeting if user types hi in multiline code block",
-    () => this.bot.usersInput([{
+  it('Shouldn\'t return any greeting if user types a word that includes greeting word', async () => {
+    await this.controller.usersInput([{
+      type: 'message',
+      channel: 'channel',
       messages: [{
-        text: '```hi```', isAssertion: true,
+        text: 'they', isAssertion: true,
       }],
     }]).then(message => assert.deepEqual(message, {}))
```

Comment:
thoughts about moving this method to a helper method for tests? It's duplicated lot of times

Reply:
@AleksSenkou Do you mean `message => assert.deepEqual(message, {})` only or the whole `usersInput`?

Reply:
@nikitasakov I'm thinking about:
```js
    await userInputs('they').then(message => assert.deepEqual(message, {}))
```

thoughts?

Reply:
Why do you use await on promise?

Reply:
or:

```js
await userInputs('they').then(noAnswerFromBotExpected)
```

Reply:
@lazycoder9 I did as in docs - https://github.com/gratifyguy/botkit-mock